http_server/request_testing/request_listener.go

```diff
@@ -0,0 +1,54 @@
+package requesttesting
+
+import (
+	"net"
+	"errors"
+	"log"
+)
+
+type mockListener struct {
+	connChannel chan net.Conn
+	serverEndpoint net.Conn
+}
+
+// For each request, we create an HTTP listener that passes the request provided
```

Comment:
nit: Document what the function actually does, not how one specific usage looks like.